qcodes/instrument_drivers/Aviosys/IP_Power_9258S.py

```diff
@@ -0,0 +1,92 @@
+from qcodes import Instrument, InstrumentChannel, ChannelList
+from qcodes.utils.validators import Enum
+from qcodes.utils.helpers import create_on_off_val_mapping
+import urllib
+
+
+class PowerChannel(InstrumentChannel):
+    """
+    Channel class for a socket on the Aviosys IP Power 9258S.
+    Args:
+        parent (Instrument): Parent instrument.
+        name (str): Channel name.
+        channel (str): Alphabetic channel id.
+    Attributes:
+        channel (str): Alphabetic channel id.
+        channel_id (int): Numeric channel id.
+    """
+
+    _channel_values = Enum('A', 'B', 'C', 'D')
+    _channel_ids = {'A': 1, 'B': 2, 'C': 3, 'D': 4}
+
+    def __init__(self, parent, name, channel):
+
+        super().__init__(parent, name)
+
+        # validate the channel id
+        self._channel_values.validate(channel)
+        self.channel = channel
+        self.channel_id = self._channel_ids[channel]
+
+        # add parameters
+        self.add_parameter('power',
+                           get_cmd=self.get_power,
+                           set_cmd=self.set_power,
+                           get_parser=int,
+                           val_mapping=create_on_off_val_mapping(on_val=1, off_val=0),
+                           label='power {}'.format(self.channel))
+
+    # get methods
+    def get_power(self):
+        request = urllib.request.Request(self.parent.address+'/set.cmd?cmd=getpower')
+        response = urllib.request.urlopen(request)
+        request_read = response.read()
+        return request_read.decode("utf-8")[4+int(self.channel_id)*6]
+
+    # set methods
+    def set_power(self, power):
+        request = urllib.request.Request(self.parent.address+'/set.cmd?cmd=setpower+p6%s=%s' % (self.channel_id, power))
```

Comment:
there is already quite some code duplication regarding `urllib.request.Request(...` etc. This can be drastically improved by defining helper functions in the instrument class which just take the important part of the URL as an argument.

One direction that can be taken is to override `ask_raw` and `write_raw` methods of the instrument driver class, so that they take care of the `urllib` business, and prepending `self.parent.address` to the URL, and other common stuff. Once that is done, the code of `set_power` on this channel will look much simpler, smth along the following lines:
```python
def set_power(self, power):
    self.write(cmd='setpower', args=f'+p6{self.channel_id}={power}')
```
which is much sorter and more readable. The `urllib` business would be then located in one place, and `set_power` code is much simpler. Together, this reduces possibilities for typos and bugs.

Moreover, if later there will be need for other parameters (don't know if this instrument has more than just "power enabled" :) ), it will be very easy to do because the `urllib` business and even the format of the URL string is taken care of by the `ask_raw`/`write_raw` methods of the instrument driver.